apps/tenant-management-webapp-e2e/src/integration/events/events.step.ts

```diff
@@ -175,3 +175,257 @@ Then('the user views the details of {string} under Core streams', function (stre
+    if (role == 'public') eventsObj.streamDetails(streamName).should('contain', '"publicSubscribe": true');
+    else {
+      eventsObj.streamDetails(streamName).should('contain', '"publicSubscribe": false');
+    }
+    const roles = role.split(',');
+    for (let i = 0; i < roles.length; i++) {
+      eventsObj
+        .streamDetails(streamName)
+        .invoke('text')
+        .then((roleDetails) => {
+          const subscriberRoles = roles[i].split(',')[0].trim();
+          expect(roleDetails).to.contain(subscriberRoles);
+        });
+    }
+    const events = event.split(',');
+    for (let i = 0; i < events.length; i++) {
+      eventsObj
+        .streamDetails(streamName)
+        .invoke('text')
+        .then((eventDetails) => {
+          const namespace = events[i].split(':')[0].trim();
+          const name = events[i].split(':')[1].trim();
+          expect(eventDetails).to.contain('"namespace": ' + '"' + namespace + '"');
+          expect(eventDetails).to.contain('"name": ' + '"' + name + '"');
+        });
+    }
+  }
+);
+
+When('the user clicks {string} button of {string}', function (button, streamName) {
+  switch (button) {
+    case 'Eye':
+      eventsObj.streamDetailsEyeIcon(streamName).click();
+      break;
+    case 'Eye-Off':
+      eventsObj.streamDetailsEyeOffIcon(streamName).click();
+      break;
+    case 'Edit':
+      eventsObj.streamEditBtn(streamName).click();
+      break;
+    case 'Delete':
+      eventsObj.streamDeleteBtn(streamName).click();
+      break;
+    default:
+      expect(button).to.be.oneOf(['Eye', 'Edit', 'Delete']);
+  }
+});
+
+Then('the user views Edit stream modal', function () {
+  eventsObj.streamModalTitle().invoke('text').should('eq', 'Edit stream');
+});
+
+Then('the user enters {string}, {string}, {string} in Edit stream modal', function (description, event, role) {
+  eventsObj.streamModalDescriptionInput().clear().type(description);
+  if (event == 'n/a') {
+    eventsObj.streamModalEventDropdown().should('exist');
+  } else {
+    const events = event.split(',');
+    eventsObj.streamModalEventDropdown().click();
+    eventsObj
+      .streamModalEventDropdownItems()
+      .then((elements) => {
+        for (let i = 0; i < elements.length; i++) {
+          if (elements[i].className.includes('goa-dropdown0-option--selected')) {
+            elements[i].click();
+          }
+        }
+      })
+      .then(() => {
+        for (let i = 0; i < events.length; i++) {
+          eventsObj.streamModalEventDropdownItem(events[i].trim()).click();
+        }
+      });
+  }
+  if (role == 'public') {
+    eventsObj
+      .streamModalPublicCheckbox()
+      .invoke('attr', 'class')
+      .then((classAttr) => {
+        if (classAttr?.includes('-selected')) {
+          cy.log('Make stream public is already checked off.');
+        } else {
+          eventsObj.streamModalPublicCheckbox().click();
+        }
+      });
+    if (role == 'n/a') {
```

Comment:
Should be "else if" instead of "if"

Reply:
Not sure which one, it gives me red underlining